specs/altair/beacon-chain.md

```diff
@@ -295,15 +297,21 @@ def get_sync_committee(state: BeaconState, epoch: Epoch) -> SyncCommittee:
     return SyncCommittee(pubkeys=pubkeys, pubkey_aggregates=pubkey_aggregates)
 ```
 
+#### `get_base_reward_per_increment`
+
+```python
+def get_base_reward_per_increment(state: BeaconState) -> Gwei:
+    return Gwei(EFFECTIVE_BALANCE_INCREMENT * BASE_REWARD_FACTOR // integer_squareroot(get_total_active_balance(state)))
```

Comment:
`EFFECTIVE_BALANCE_INCREMENT * BASE_REWARD_FACTOR = 2**0 * 10**9 * 64 = 64 * 10**9`, so 35.89... bits if dividing by 1. (super low total balance case). 64-36=28 of room. Simple multiplication with increment count will work, something like squaring won't.

Reply:
Quite an extreme case. Is there some minimum total balance bit count we can assume here? Worth documenting the overflow. A year ago we Herman found a similar thing in the original code.

Reply:
Oh right, I remember now, the total active balance has a lower bound of 1 effective balance increment. `log2(1 * 10**9) = 29.8...`. That should leave it at ~6 bits, low.